spring-boot-project/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json

```diff
@@ -2133,6 +2133,64 @@
       "name": "spring.neo4j.uri",
       "defaultValue": "bolt://localhost:7687"
     },
+    {
+      "name": "spring.pulsar.consumer.dead-letter-policy.dead-letter-topic",
+      "type": "java.lang.String",
+      "description": "Name of the dead topic where the failing messages will be sent."
+    },
+    {
+      "name": "spring.pulsar.consumer.dead-letter-policy.initial-subscription-name",
+      "type": "java.lang.String",
+      "description": "Name of the initial subscription name of the dead letter topic - must be set for the initial subscription for the dead letter topic to be created."
+    },
+    {
+      "name": "spring.pulsar.consumer.dead-letter-policy.max-redeliver-count",
+      "type": "java.lang.Integer",
+      "description": "Maximum number of times that a message will be redelivered before being sent to the dead letter queue."
+    },
+    {
+      "name": "spring.pulsar.consumer.dead-letter-policy.retry-letter-topic",
+      "type": "java.lang.String",
+      "description": "Name of the retry topic where the failing messages will be sent."
+    },
+    {
+      "name": "spring.pulsar.function.enabled",
+      "type": "java.lang.Boolean",
+      "description": "Whether to enable function support.",
+      "defaultValue": true
+    },
+    {
+      "name": "spring.pulsar.producer.cache.enabled",
+      "type": "java.lang.Boolean",
+      "description": "Whether to enable caching in the PulsarProducerFactory.",
+      "defaultValue": true
+    },
+    {
+      "name": "spring.pulsar.reactive.consumer.dead-letter-policy.dead-letter-topic",
+      "type": "java.lang.String",
+      "description": "Name of the dead topic where the failing messages will be sent."
+    },
+    {
+      "name": "spring.pulsar.reactive.consumer.dead-letter-policy.initial-subscription-name",
+      "type": "java.lang.String",
+      "description": "Name of the initial subscription name of the dead letter topic - must be set for the initial subscription for the dead letter topic to be created."
+    },
+    {
+      "name": "spring.pulsar.reactive.consumer.dead-letter-policy.max-redeliver-count",
+      "type": "java.lang.Integer",
+      "description": "Maximum number of times that a message will be redelivered before being sent to the dead letter queue."
+    },
+    {
+      "name": "spring.pulsar.reactive.consumer.dead-letter-policy.retry-letter-topic",
+      "type": "java.lang.String",
+      "description": "Name of the retry topic where the failing messages will be sent."
+    },
+    {
+      "name": "spring.pulsar.reactive.sender.cache.enabled",
```

Comment:
Where does this come from? The `Sender` class doesn't have a `Cache`.

Reply:
It is hiding in the plethora of other properties ([here](https://github.com/spring-projects/spring-boot/pull/34763/files#diff-9ae54aecfff5dde08e706cda1fda7a45880dcee70f9e3f07f8ab460026d452f8R413) and [defined here](https://github.com/spring-projects/spring-boot/pull/34763/files#diff-9ae54aecfff5dde08e706cda1fda7a45880dcee70f9e3f07f8ab460026d452f8R226)).

Is it best practice to declare the nested props at the top of the containing class? I think so, and by moving the 
```java
private final Cache cache = new Cache();
```
to the top of `Sender` class would make this more readable. Wdyt?